applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -263,6 +269,76 @@ static bool app_event_handler(const struct app_event_header *aeh)
 	return false;
 }
 
+static bool grant_send(enum coneval_supported_data_type type)
+{
+#if defined(CONFIG_DATA_GRANT_SEND_ON_CONNECTION_QUALITY)
+	int err;
+	struct lte_lc_conn_eval_params coneval = { 0 };
+
+	if (send_denied_count[type] >= CONFIG_DATA_SEND_ATTEMPTS_COUNT_MAX) {
+		/* Grant send if a message has been attempted too many times. */
+		LOG_WRN("Too many attempts, granting send");
+		goto grant;
+	}
+
+	err = lte_lc_conn_eval_params_get(&coneval);
+	if (err < 0) {
+		LOG_ERR("lte_lc_conn_eval_params_get, error: %d", err);
+		SEND_ERROR(cloud, CLOUD_EVT_ERROR, err);
+	} else if (err > 0) {
+		LOG_WRN("Connection evaluation failed, error: %d", err);
+		goto grant;
+	}
+
+	LOG_DBG("Current LTE energy estimate: %d", coneval.energy_estimate);
+
+	switch (type) {
+	case GENERIC:
+		if (coneval.energy_estimate >= LTE_LC_ENERGY_CONSUMPTION_NORMAL) {
+			goto grant;
+		}
+		break;
+	case NEIGHBOR_CELLS:
+		if (coneval.energy_estimate >= LTE_LC_ENERGY_CONSUMPTION_NORMAL) {
+			goto grant;
+		}
+		break;
+	case BATCH: {
+		enum lte_lc_energy_estimate estimate_min;
+
+		/* The nRF Cloud integration uses batch data exclusively for regular data updates.
+		 * Due to this we lower the energy estimate requirement for batch data type.
+		 */
+		if (IS_ENABLED(CONFIG_NRF_CLOUD_MQTT)) {
+			estimate_min = LTE_LC_ENERGY_CONSUMPTION_NORMAL;
+		} else {
+			estimate_min = LTE_LC_ENERGY_CONSUMPTION_EFFICIENT;
```

Comment:
Same, I think it can be changed to LTE_LC_ENERGY_CONSUMPTION_REDUCED for this initial implementation. Ill make sure to add kconfig choices for NCELL, GENERIC, and BATCH types so that the user can set their desired level.